peps/pep-0787.rst

```diff
@@ -0,0 +1,279 @@
+PEP: 787
+Title: Safer subprocess usage using t-strings
+Author: Alyssa Coghlan <ncoghlan@gmail.com>, Nick Humrich <nick@humrich.us>
+Discussions-To: Pending
+Status: Draft
+Type: Standards Track
+Requires: 750
+Created: 11-Apr-2025
+Python-Version: 3.14
```

Comment:
Time is short:

The beta feature freeze is in just over 3 weeks, and in that time this PR needs merging (we can do that quickly), then you need to open and guide the discussion thread, potentially update the PEP and iterate, submit to the SC, await their decision, then open a PR with changes to CPython, and have that reviewed and merged before 6th May. 

Therefore I suggest you work on the implementation in parallel. Note that PEP 750 hasn't been merged yet, or even a PR opened, so you'll need to branch off the reference implentation branch (linked in the PEP).

Reply:
I suspect if we're going to hit controversy, it's going to be in relation to the `shell=True` behaviour with non-POSIX shells (there are good reasons the PEP 750 authors declined to make this shell quoting proposal part of the base t-string PEP). If PEP 787's position of "it's still markedly better than the status quo, even though it doesn't fix everything everywhere" isn't accepted, then yeah, this is going to slip to 3.15.

I'm not overly concerned about the implementation side of things - the quoted code in the PEP really is pretty much everything the reference implementation needs (aside from test case additions).

Reply:
I suggest expecting 3.15 for this one as APIs with security implications require thought. 

Even if it is markedly better than the status quo if a behavior is determined to be wrong or unsupportable once this ships, it becomes a breaking change for someone if we fix the behavior in the future.

Reply:
FWIW even if not fully plumbed through subprocess yet, I think allowing `shlex.join` to take a Template as input would be an interesting stdlib improvement on its own as maybe something to do before this whole PEP lands?  something to ponder.

Reply:
I don't understand the usefulness of adding Template support to join. Could you provide an example perhaps?